atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/DefaultClusterEventService.java

```diff
@@ -189,6 +190,10 @@ private String topicsAsString(final Set<String> topics) {
           Executors.newSingleThreadScheduledExecutor(
               namedThreads("atomix-cluster-event-executor-%d", LOGGER));
       membershipService.addListener(this);
+      // Listener doesn't receive notification about the Members added before the listener is added.
+      membershipService
+          .getMembers()
+          .forEach(m -> event(new ClusterMembershipEvent(Type.MEMBER_ADDED, m)));
```

Comment:
might make sense to do that in the membershipService#addListener? Because this problem will also arise on other places probably furthermore the membership is in charge to do that I would say.

Reply:
That's right. It would prevent similar bugs in future. It is not the first time we encountered this I guess. That said, is it ok if I do it in another PR? We have to verify if it breaks existing listeners.

Reply:
Yes then please create a follow up issue :slightly_smiling_face: